### PR TITLE
Add setup code for API call metrics for sync

### PR DESCRIPTION
**Pull request by dagnir**
## Description
Part of Implementing core metrics. This PR adds code to set up the metric collection and publishing for sync API calls.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## Checklist

- [ ] I have read the **CONTRIBUTING** document
- [ ] Local run of `mvn install` succeeds
- [ ] My code follows the code style of this project
- [ ] My change requires a change to the Javadoc documentation
- [ ] I have updated the Javadoc documentation accordingly
- [ ] I have read the **README** document
- [ ] I have added tests to cover my changes
- [ ] All new and existing tests passed
- [ ] A short description of the change has been added to the **CHANGELOG**
- [ ] My change is to implement 1.11 parity feature and I have updated [LaunchChangelog](https://github.com/aws/aws-sdk-java-v2/blob/master/docs/LaunchChangelog.md)

## License

- [x] I confirm that this pull request can be released under the Apache 2 license

### CODE REVIEW

File: core/aws-core/src/main/java/software/amazon/awssdk/awscore/metrics/AwsCoreMetric.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.awscore.metrics;
+
+import software.amazon.awssdk.annotations.SdkPublicApi;
+import software.amazon.awssdk.metrics.MetricCategory;
+import software.amazon.awssdk.metrics.SdkMetric;
+
+@SdkPublicApi
```
Comment (bmaizels) [inline core/aws-core/src/main/java/software/amazon/awssdk/awscore/metrics/AwsCoreMetric.java:22]: Would protected be a better designation?

Comment (dagnir) [reply core/aws-core/src/main/java/software/amazon/awssdk/awscore/metrics/AwsCoreMetric.java:22]: These object are potentially accessed by custom-provided metric publishers so these need to be designated as public

File: core/aws-core/src/main/java/software/amazon/awssdk/awscore/metrics/AwsCoreMetric.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+import software.amazon.awssdk.annotations.SdkPublicApi;
+import software.amazon.awssdk.metrics.MetricCategory;
+import software.amazon.awssdk.metrics.SdkMetric;
+
+/**
+ * The set of metrics collected for all SDK client operations.
+ */
+@SdkPublicApi
+public final class AwsCoreMetric {
+    public static final SdkMetric<String> SERVICE_ID = metric("ServiceName", String.class);
```
Comment (bmaizels) [inline core/aws-core/src/main/java/software/amazon/awssdk/awscore/metrics/AwsCoreMetric.java:27]: These need javadocs too...

Comment (dagnir) [reply core/aws-core/src/main/java/software/amazon/awssdk/awscore/metrics/AwsCoreMetric.java:27]: Oops! will fix

**Pull request merged**